theano/gpuarray/opt.py

```diff
@@ -215,6 +256,248 @@ def apply(self, fgraph):
                     0, 'fast_run', 'fast_compile', 'merge')
 
 
+class GraphToGPU(NavigatorOptimizer):
+    """
+    Transfer the graph as a whole to GPU instead of transfering node by node.
+    """
+
+    def __init__(self, local_optimizers_all, local_optimizers_map):
+        self.local_optimizers_all = local_optimizers_all
+        self.local_optimizers_map = local_optimizers_map
+        self.failure_callback = None
+
+    def add_requirements(self, fgraph):
+        fgraph.attach_feature(toolbox.ReplaceValidate())
+
+    def apply(self, fgraph):
+        mapping = {}
+        time_opts = {}
+        node_created = {}
+        process_count = {}
+        t_topo = time.time()
+        topo = fgraph.toposort()
+        time_topo = time.time()
+        toposort_timing = time_topo - t_topo
+
+        # Building a new graph
+        # Iterating through inputs of graph
+        target = infer_context_name(*fgraph.inputs)
```

Comment:
This fixed me the error that i've reported previously. https://github.com/Theano/Theano/pull/4570#discussion_r67513551